core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -479,7 +479,7 @@ public void run()
                         return;
                     }
 
-                    String threadId = split.getTaskHandle().getTaskId() + "-" + split.getSplitId();
+                    String threadId = split.getTaskHandle().getTaskId() + "-" + split.getSplitId() + "-" + System.nanoTime();
```

Comment:
This may negatively impact tools that gather thread dumps and attempt to group threads by name for analysis purposes.